src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.cs

```diff
@@ -422,10 +404,37 @@ public SolutionState RemoveProject(ProjectId projectId)
                 dependencyGraph: newDependencyGraph);
         }
 
+        public ImmutableDictionary<string, ImmutableArray<DocumentId>> CreateFilePathToDocumentIdsMapWithAddedDocuments(IEnumerable<TextDocumentState> documentStates)
+        {
+            var builder = _filePathToDocumentIdsMap.ToBuilder();
+            AddDocumentFilePaths(documentStates, builder);
+            return builder.ToImmutable();
+        }
+
+        private static void AddDocumentFilePaths(IEnumerable<TextDocumentState> documentStates, ImmutableDictionary<string, ImmutableArray<DocumentId>>.Builder builder)
```

Comment:
other pr wants to be able to call directly into this (and RemoveDocumentFilePaths).